src/CoreWCF.Http/src/CoreWCF/Channels/BasicHttpSecurityMode.cs

```diff
@@ -0,0 +1,42 @@
+﻿namespace CoreWCF.Channels
+{
+    public enum BasicHttpSecurityMode
+    {
+        None,
+        Transport,
+        Message,
+        TransportWithMessageCredential,
+        TransportCredentialOnly
+    }
+
+    static class BasicHttpSecurityModeHelper
+    {
+        internal static bool IsDefined(BasicHttpSecurityMode value)
+        {
+            return (value == BasicHttpSecurityMode.None ||
```

Comment:
This code is to prevent something like this:
```c#
    basicHttpBinding.Security.Mode = (BasicHttpSecurityMode)27;
```
It's not used for validating whether the requested binding is configured in a usable way, it's there to prevent a garbage value being set on the property as enum's are one step above semantic sugar around an int.